gbpservice/neutron/services/servicechain/drivers/oneconvergence_intercloud_driver.py

```diff
@@ -241,12 +271,189 @@ def _fetch_template_and_params(self, context, sc_instance,
             for parameter in config_param_values.keys():
                 if parameter in node_params.keys():
                     stack_params[parameter] = config_param_values[parameter]
-        LOG.debug(stack_template)
+        LOG.debug(_("Final stack_template = %s, stack_params = %s"), stack_template, stack_params)
 
         return stack_template, stack_params
 
-    def _update_template_with_firewall_rules(self, context, provider_ptg,
-                                             provider_cidr, consumer_cidr,
+    def _get_hotplug_port_id(self, context, subnet_id):
+        """
+        First remove the router interface.
+        Then create a new port with the same ip address, and hotplug
+        it to the provided instance
+        """
+        hotplug_port = None
+        subnet = self._core_plugin.get_subnet(context._plugin_context, subnet_id)
+        network_id = subnet['network_id']
+        ports = self._core_plugin.get_ports(
+            context._plugin_context,
+            filters={'device_owner': ['network:router_interface']})
+        router_id = None
+        if ports:
+            for port in ports:
+               if port['fixed_ips'][0]['subnet_id'] == subnet_id:
```

Comment:
Not sure if its possible, but please check if the fixed_ips can also be used in filters to avoid this loop here.